components/script/dom/request.rs

```diff
@@ -0,0 +1,857 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+use dom::bindings::cell::DOMRefCell;
+use dom::bindings::codegen::Bindings::RequestBinding;
+use dom::bindings::codegen::Bindings::RequestBinding::ReferrerPolicy;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestCache;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestCredentials;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestDestination;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestInfo;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestInit;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestMethods;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestMode;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestRedirect;
+use dom::bindings::codegen::Bindings::RequestBinding::RequestType;
+use dom::bindings::codegen::UnionTypes::{HeadersOrByteStringSequenceSequence, RequestOrUSVString};
+use dom::bindings::error::{Error, Fallible};
+use dom::bindings::global::GlobalRef;
+use dom::bindings::js::{JS, MutNullableHeap, Root};
+use dom::bindings::reflector::{Reflectable, Reflector, reflect_dom_object};
+use dom::bindings::str::{ByteString, USVString, DOMString};
+use dom::headers::{Headers, Guard};
+use hyper;
+use msg;
+use net_traits::request as NetTraitsRequest;
+use std::cell::{Cell, Ref, RefCell};
+use url;
+
+#[dom_struct]
+pub struct Request {
+    reflector_: Reflector,
+    #[ignore_heap_size_of = "net_traits is missing HeapSizeOf implementation"]
+    request: DOMRefCell<NetTraitsRequest::Request>,
+    body_used: DOMRefCell<bool>,
+    headers_reflector: MutNullableHeap<JS<Headers>>,
+    mime_type: DOMRefCell<Vec<u8>>,
+}
+
+impl Request {
+    pub fn new_inherited(url: url::Url,
+                         origin: Option<NetTraitsRequest::Origin>,
+                         is_service_worker_global_scope: bool,
+                         pipeline_id: Option<msg::constellation_msg::PipelineId>) -> Request {
+        Request {
+            reflector_: Reflector::new(),
+            request: DOMRefCell::new(
+                NetTraitsRequest::Request::new(url,
+                                               origin,
+                                               is_service_worker_global_scope,
+                                               pipeline_id)),
+            body_used: DOMRefCell::new(false),
+            headers_reflector: Default::default(),
+            mime_type: DOMRefCell::new("".to_string().into_bytes()),
+        }
+    }
+
+    pub fn new(global: GlobalRef,
+               url: url::Url,
+               origin: Option<NetTraitsRequest::Origin>,
+               is_service_worker_global_scope: bool,
+               pipeline_id: Option<msg::constellation_msg::PipelineId>) -> Root<Request> {
```

Comment:
`use msg::constellation_msg::PipelineId` above and you can simply use `PipelineId` here.